awx/ui_next/src/screens/Inventory/InventorySources/InventorySourceList.jsx

```diff
@@ -105,8 +105,33 @@ function InventorySourceList({ i18n }) {
   );
   const { error: syncError, dismissError } = useDismissableError(syncAllError);
 
+  const deleteRelatedInventoryResources = resourceId => {
```

Comment:
the use of async is confusing here.  it doesn't look like you're returning anything from this function.  so selected.map would return an array of undefineds...so the Promise.all below will never resolve  (or will instantly resolve regardless of if the functions are  called.  I believe this will work correctly:

```
const deleteRelatedInventoryResources = resourceId => {
    return [InventorySourcesAPI.destroyHosts(resourceId), 
    InventorySourcesAPI.destroyGroups(resourceId)];
  };

  const {
    isLoading: deleteRelatedResourcesLoading,
    deletionError: deleteRelatedResourcesError,
    deleteItems: handleDeleteRelatedResources,
  } = useDeleteItems(
    useCallback(async () => {
      return (
        Promise.all(
          selected.map(({ id: resourceId }) =>
            deleteRelatedInventoryResources(resourceId)
          ).flat();
          console.log("you'll get here after everything is resolved");
        ),
        []
      );
    }, [selected])
  );
```

Reply:
@jlmitch5, thanks for your review. Updating now.